src/lib/encoder.ts

```diff
@@ -143,6 +143,109 @@ const decodeCompactBytesArray = (compactBytesArray: string): string[] => {
   return encodedValues;
 };
 
+const encodeBytesNCompactBytesArray = (
+  values: string[],
+  numberOfBytes: number,
+): string => {
+  values.forEach((value, index) => {
+    if (value.length > numberOfBytes * 2 + 2)
+      throw new Error(
+        `Hex bytes${numberOfBytes} value at index ${index} does not fit in ${numberOfBytes} bytes`,
+      );
+  });
+
+  return encodeCompactBytesArray(values);
+};
+
+const decodeBytesNCompactBytesArray = (
+  compactBytesArray: string,
+  numberOfBytes: number,
+): string[] => {
+  const bytesValues = decodeCompactBytesArray(compactBytesArray);
+  bytesValues.forEach((bytesValue, index) => {
+    if (bytesValue.length > numberOfBytes * 2 + 2)
+      throw new Error(
+        `Hex bytes${numberOfBytes} value at index ${index} does not fit in ${numberOfBytes} bytes`,
+      );
+  });
+
+  return bytesValues;
+};
+
+const returnTypesOfBytesNCompactBytesArray = () => {
+  const types = {};
+  for (let i = 1; i < 33; i++) {
```

Comment:
Can't it go higher than 32? as between each value we have a bytes2 (65535)

Reply:
I can but then it's not a valid Solidity type, higher than 32 is just bytes.

Reply:
Alright I see